File: CSharp/LeetCode/leetCode359/p359.cs
public class Logger {

    Dictionary<string,int> map;
    public Logger() {
        map = new Dictionary<string,int>();
    }
    
    public bool ShouldPrintMessage(int timestamp, string message) {
        int span = 10;

        if(map.ContainsKey(message) && timestamp - map[message] < span){
            return false;
        }
        map[message] = timestamp;
        return true;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * bool param_1 = obj.ShouldPrintMessage(timestamp,message);
 */

File: CSharp/LeetCode/leetCode359/p359.ts
class Logger {
    map:Map<string,number>;
    
    constructor() {
        this.map = new Map();
    }

    shouldPrintMessage(timestamp: number, message: string): boolean {
        let span = 10;
        if(this.map.has(message) && timestamp - this.map.get(message) < span) {
            return false;
        }
        this.map.set(message,timestamp);
        return true;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * var obj = new Logger()
 * var param_1 = obj.shouldPrintMessage(timestamp,message)
 */